src/mon/MgrStatMonitor.cc

```diff
@@ -69,6 +103,29 @@ void MgrStatMonitor::create_initial()
 void MgrStatMonitor::calc_pool_availability()
 {
   dout(20) << __func__ << dendl;
+  std::scoped_lock l(lock);
```

Comment:
Wasn't this lock removed?  I actually think you need it.  handle_conf_change iirc is called from a thread created by  the config subsystem, so accessing enable_availability_tracking, reset_availability_last_update_downtime_val from both places does need a lock.  However, you'd also need to take the lock in the if block in handle_conf_change.